libshared/RandomToolset.h

```diff
@@ -0,0 +1,170 @@
+/**
+ * \section License
+ * This file is part of JuPedSim.
+ *
+ * JuPedSim is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * any later version.
+ *
+ * JuPedSim is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with JuPedSim. If not, see <http://www.gnu.org/licenses/>.
+ **/
+#pragma once
+
+#include <algorithm>
+#include <assert.h>
+#include <random>
+#include <stdexcept>
+
+/**
+ * Toolset for creating random numbers
+ */
+namespace Random
+{
+namespace
+{
+/**
+ * Random engine used for creating the random numbers
+ */
+std::mt19937 _randomEngine;
+
+/**
+ * Flag if a seed was properly set
+ */
+bool _init = false;
```

Comment:
Should be `std::atomic_flag` or `std::atomic<bool>`

Reply:
Changed